backend/utils/advanceQueue.js

```diff
@@ -0,0 +1,34 @@
+// const dynamicBuffer = require('./dynamicBuffer'); // Import dynamicBuffer 
+// const sendWebNotification = require('./sendWebNotification'); // Import sendWebNotification 
+
+async function advanceQueue(locationData) {
+    const { activeFirebaseUIDs, activeNicknames, activeStartTimes, activeWaitingPlayers, queueFirebaseUIDs, queueNicknames } = locationData;
+
+    // If there are no players in the queue, return early (don't advance the queue)
+    if (queueFirebaseUIDs.length === 0) {
+        return;
+    }
+
+    // Get empty court index, and validate
+    const emptyCourtIdx = activeFirebaseUIDs.findIndex((firebaseUID) => firebaseUID.startsWith('Empty'));
+    // if there are no empty courts, emptyCourtIdx will be -1 (don't advance the queue)
+    if (emptyCourtIdx === -1) { 
+        return;
+    }
+
+    // Move first player in the queue to active
+    const firstPlayerUID = queueFirebaseUIDs.shift();
+    const firstPlayerNickname = queueNicknames.shift();
+    activeFirebaseUIDs[emptyCourtIdx] = firstPlayerUID; 
+    activeNicknames[emptyCourtIdx] = firstPlayerNickname;
+    activeStartTimes[emptyCourtIdx] = new Date();
+    activeWaitingPlayers[emptyCourtIdx] = false;    
+
+    // Call dynamicBuffer 
+    // await dynamicBuffer(locationData);
+
+    // Notify second player that they are next
+    // await sendWebNotification(locationData, secondPlayer);
```

Comment:
I believe we should only send a web notification if the queue is not empty, i.e. there is someone to send the web notification to (after the queue has successfully advanced once)? It also might be more straightforward to replace secondPlayer with the first element of the queueFirebaseUIDs (given that it's non-empty) so the person implementing the sendWebNotification endpoint can uncomment the line without writing the logic themselves :).

Reply:
Also, I'm pretty sure sendWebNotification only takes in a firebaseUID and a message, so you don't need to pass in the locationData 👍

Reply:
That's a good point, thanks for the feedback! I updated the code with "next player" instead of "second player" and changed the sendWebNotification's call logic. Feel free to let me know if you have any other suggestions!